lib/driverProviders/browserstack.js

```diff
@@ -0,0 +1,87 @@
+/*
+ * This is an implementation of the Browserstack Driver Provider.
+ * It is responsible for setting up the account object, tearing
+ * it down, and setting up the driver correctly.
+ */
+
+var util = require('util'),
+    log = require('../logger.js'),
+    request = require('request'),
+    q = require('q'),
+    DriverProvider = require('./driverProvider');
+
+
+var BrowserStackDriverProvider = function(config) {
+  DriverProvider.call(this, config);
+};
+util.inherits(BrowserStackDriverProvider, DriverProvider);
+
+
+/**
+ * Hook to update the BrowserStack job status.
+ * @public
+ * @param {Object} update
+ * @return {q.promise} A promise that will resolve when the update is complete.
+ */
+BrowserStackDriverProvider.prototype.updateJob = function(update) {
+
+  var self = this;
+  var deferredArray = this.drivers_.map(function(driver) {
+    var deferred = q.defer();
```

Comment:
I saw this documented [here](https://github.com/angular/protractor/blob/master/lib/driverProviders/README.md)

I guess here `this.drivers_` is an array and `updateJob` should return only after all the 'drivers' have returned.